master/internal/rm/kubernetesrm/spec.go

```diff
@@ -557,6 +571,18 @@ func configureSecurityContext(agentUserGroup *model.AgentUserGroup) *k8sV1.Secur
 	return nil
 }
 
+func configureDetContainerSecurityContext(
+	agentUserGroup *model.AgentUserGroup,
+	securityContext *k8sV1.SecurityContext,
+) *k8sV1.SecurityContext {
+	context := configureSecurityContext(agentUserGroup)
+	if context != nil {
+		securityContext.RunAsUser = context.RunAsGroup
+		securityContext.RunAsGroup = context.RunAsGroup
+	}
```

Comment:
to be conservative, if context is nil, just erase the settings in the passed in securityContext.

That's easier to document too: "we don't honor user/group in podspec security context, use link-with-agent-user instead"